Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -119,6 +122,33 @@ class TabScrollingController: NSObject {
             }
         }
     }
+
+    func updateMinimumZoom() {
+        guard let scrollView = scrollView else {
+            return
+        }
+        self.isZoomedOut = roundNum(scrollView.zoomScale) == roundNum(scrollView.minimumZoomScale)
+        self.lastZoomedScale = self.isZoomedOut ? 0 : scrollView.zoomScale
+    }
+
+    func setMinimumZoom() {
+        guard let scrollView = scrollView else {
+            return
+        }
+        if self.isZoomedOut && roundNum(scrollView.zoomScale) != roundNum(scrollView.minimumZoomScale) {
+            scrollView.zoomScale = scrollView.minimumZoomScale
+        }
+    }
+
+    func resetZoomState() {
+        self.isZoomedOut = false
+        self.lastZoomedScale = 0
+    }
+
+    private func roundNum(num: CGFloat) -> CGFloat {
+        return round(100000 * num) / 100000
```

Comment:
Just curious, what's the purpose of doing this? Is it so we can have a more accurate round?